datafusion-cli/src/print_options.rs

```diff
@@ -103,4 +109,28 @@ impl PrintOptions {
 
         Ok(())
     }
+
+    #[allow(clippy::println_empty_string)]
+    pub async fn print_stream(
+        &self,
+        mut stream: Pin<Box<dyn RecordBatchStream>>,
+        query_start_time: Instant,
+    ) -> Result<()> {
+        let mut row_count = 0_usize;
+        let mut with_header = true;
```

Comment:
I wonder if there is any way to test this behavior (specifically that the header is only printed out for the first batch) so that it isn't broken in subsequent refactorings